x/pairing/keeper/scores/score.go

```diff
@@ -189,17 +188,10 @@ func PickProviders(ctx sdk.Context, scores []*PairingScore, groupCount int, hash
 		return returnedProviders
 	}
 
-	sum256 := sha256.Sum256(hashData)
-	// Fold the SHA-256 hash into a 64-bit seed using bitwise XOR
-	var seed int64
-	for i := 0; i < len(sum256)/8; i++ {
-		seed ^= int64(binary.BigEndian.Uint64(sum256[i*8 : (i+1)*8]))
-	}
-
-	rand.Seed(seed)
+	rng := rand.New(hashData)
 
 	for it := 0; it < groupCount; it++ {
-		randomValue := uint64(rand.Int63n(scoreSum.BigInt().Int64())) + 1
+		randomValue := uint64(rng.Int63n(scoreSum.BigInt().Int64())) + 1
```

Comment:
the +1 needs to be inside the random

Reply:
the +1 is at the right place as it is.
we want a random number between 1 and the scoreSum, as values 1-score-of-first-item would mean "pick the first", and the next score-of-second-item values would mean "pick the second" and so on. but `Int63n()` returns a number between 0 and scoreSum-1 - so we need to shift up by one.